ceno_zkvm/src/structs.rs

```diff
@@ -152,10 +152,10 @@ impl<E: ExtensionField> ZKVMConstraintSystem<E> {
         config
     }
 
-    pub fn register_table_circuit<TC: TableCircuit<E>>(&mut self) -> TC::TableConfig {
+    pub fn register_table_circuit<TC: TableCircuit<E>>(&mut self, tc: TC) -> TC::TableConfig {
```

Comment:
There could be a convenience method (original name) that uses Default (if implemented) and a new one with the parameter.

```rust
impl<TC> …
where TC::TableConfig: Default {
    pub fn register_table_circuit(…) {
        let tc = Default::default();
…
}

impl<TC> … {
    pub fn register_table_circuit_param(…, tc: TC) {
…
}
```

Reply:
This way there is no need to change every call.

Reply:
:+1:  agreed, got a bit carried away there